src/inmanta/data/model.py

```diff
@@ -749,12 +749,14 @@ class LoginReturn(BaseModel):
     user: User
 
 
-def is_resource_id(v: Optional[str], info: ValidationInfo, allow_none: bool = False) -> str | None:
-    if v is None:
-        if allow_none:
-            return v
-        raise ValidationError(f"{info.field_name} is None")
-    return resources.Id.parse_id(v).resource_str()
+def is_resource_id(v: ResourceIdStr, info: pydantic.ValidationInfo) -> str:
+    try:
+        return resources.Id.parse_id(v).resource_str()
+    except Exception:
+        raise ValueError(f"Validation failed for {info.field_name}")
```

Comment:
The parse method already raises a `ValueError` with a more useful error message. I would drop the try-catch, perhaps with a comment that we rely on the method's `ValueError` for validation.